src/v0/util/index.js

```diff
@@ -2296,14 +2296,53 @@ const applyJSONStringTemplate = (message, template) =>
   }).evaluate(message);
 
 /**
- * This groups the events by destination ID and source ID.
+ * This groups the events by destination ID, source ID, and optionally by destination config.
  * Note: sourceID is only used for rETL events.
+ *
+ * The function checks the hasDynamicConfig flag on the destination to determine if events
+ * should be grouped by destination config as well. This is important for destinations that
+ * use dynamic configuration where different events might need different config values.
+ *
+ * For backward compatibility with older server versions where the hasDynamicConfig flag
+ * might not be available, the function treats undefined flags as if dynamic config might
+ * be present (only skips grouping by config when the flag is explicitly false).
+ *
  * @param {*} events The events to be grouped.
+ * @param {string} destinationType The destination type (for logging purposes).
  * @returns {array} The array of grouped events.
  */
-const groupRouterTransformEvents = (events) =>
+const groupRouterTransformEvents = (events, destinationType) =>
   Object.values(
-    lodash.groupBy(events, (ev) => [ev.destination?.ID, ev.context?.sources?.job_id || 'default']),
+    lodash.groupBy(events, (ev) => {
+      // Check if hasDynamicConfig flag is available
+      // If undefined (older server versions), process all events as if they might have dynamic config
+      // Only skip processing if the flag is explicitly false
+      const hasDynamicConfigFlag = ev.destination?.hasDynamicConfig;
+      const shouldGroupByDestinationConfig = hasDynamicConfigFlag !== false;
+
+      // If we should group by destination config, include it in the grouping key
+      // Otherwise, use 'default' to group all events with the same destination ID together
+      let destConfigGroupKey = 'default';
+
+      if (shouldGroupByDestinationConfig && ev.destination?.Config) {
+        // Sort the keys to ensure consistent ordering regardless of how the object was created
+        // This ensures that { a: 1, b: 2 } and { b: 2, a: 1 } are treated as the same config
+        const sortedConfig = {};
```

Comment:
we used use this https://www.npmjs.com/package/fast-json-stable-stringify instead of manual as we are not taking care of nested keys so better to offload this is known library